spec/models/doi_spec.rb

```diff
@@ -298,10 +298,31 @@
       expect(doi.language).to eq("fr")
     end
 
-    it "error" do
+    it "human longer than 8 characters" do
+      doi.language = "Indonesian"
+      expect(doi.save).to be true
+      expect(doi.errors.details).to be_empty
+      expect(doi.language).to eq("id")
+    end
+
+    it "non-iso 639-1" do
       doi.language = "hhh"
       expect(doi.save).to be true
       expect(doi.errors.details).to be_empty
+      expect(doi.language).to eq("hhh")
+    end
+
+    it "non-iso 639-1 with country code" do
+      doi.language = "prq-PE"
+      expect(doi.save).to be true
+      expect(doi.errors.details).to be_empty
+      expect(doi.language).to eq("prq-PE")
+    end
+
+    it "fails xs:language regex" do
+      doi.language = "Borgesian"
+      expect(doi.save).to be true
+      expect(doi.errors.details).to be_empty
```

Comment:
If the language saved is `nil` (i.e the input was invalid), is there a reason we are not returning an error message?

Reply:
I'm not sure. This was how it was implemented before—see the original `it "error" do` test, which is expected to return `nil` instead of an error—so I imagine there was some intention. My guess is that we didn't want to reject all records with non-ISO 639-1 codes and names in the language property since this could cause headaches for users.

Reply:
I am worried that silently dropping data is not a good idea, even if it's what we have already been doing.

Reply:
That makes sense. That said, I am concerned with changing the behavior, since this may cause issues with records that could have been submitted successfully in the past. The XSD and the model validation will block any invalid values even if they make it in.

Reply:
The current behavior has been logged as an issue here: https://github.com/datacite/lupo/issues/904